### ISSUE TITLE
file does not open

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#789]
this flora does not open. You can download it from here
https://we.tl/t-Rr3OWTpUnV

[GgImagine.20191108-0933.out.zip](https://github.com/gsautter/goldengate-imagine/files/3825597/GgImagine.20191108-0933.out.zip)

**Comment by myrmoteras** [comment]
@gsautter do you have a chance to look at this and let me know whether this is processable? It is a series from the Paris Museum and Laurence is going to the World Flora Online meeting at missour Botanical garden where she might suggest this series as an input, and pay us to convert it.

**Comment by gsautter** [comment]
I did download the file and tried to test it, but it's simply too large a PDF ... looking into finally getting a new machine asap.
As to this very issue, I'd like to try and test this on your machine when I come to Paris at the end of next week. The error shown in the log happens in a pretty weird place where I can hardly imagine something like that happening, so I have a hard time telling what might be going wrong.

In the meantime, as this error happens in the process of font decoding, you might try and run the file with "Render Glyphs Only" to bypass the problematic code sequence. Might have to correct a few characters afterwards, but that's quite possible via "Edit > Edit Embedded Fonts".

**Comment by myrmoteras** [comment]
I run this the way you recommend "render glyphs only", but now I get this
![image](https://user-images.githubusercontent.com/4609956/69917352-154f6a80-1465-11ea-85ed-a82795615ec5.png)

how long should I let it run? usese al my CPU
![image](https://user-images.githubusercontent.com/4609956/69917369-43cd4580-1465-11ea-98ec-ec454e1d3da3.png)

**Comment by myrmoteras** [comment]
hey, it's winter here and I get complemtary heating

![image](https://user-images.githubusercontent.com/4609956/69917455-9b1fe580-1466-11ea-831a-55ad4cd5db77.png)

**Comment by myrmoteras** [comment]
but here is the file now: (1.GB). Can you look at this with your expertise on whether this coudl be processed by a template, and the captions etc could be discovered, the objects left and right in the descriptions can be omitted, et c. 

https://drive.google.com/open?id=1db9d1CreyOaZ4BwtTiv9MuG16Pd-Ya3F

**Comment by gsautter** [comment]
Holy s**t ... that one surely looks like a piece of work if it got your machine laboring like that.

**Comment by myrmoteras** [comment]
You have a credit to buy you a similar machine..

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by gsautter** [comment]
Regarding the possibility of a template:
- The fact that the whole document is in French might pose a bit of a challenge, as  pretty much all our heading classification is tuned to English keywords, but that's something we can deal with.
- The headings proper are nice and regular, very good for templating.
- The reference groups (e.g. for Cunoniaceae on Page 10/455, or Cunonia atrorubens on Page 122/455) might be a bit of a challenge for templates due to the narrow-column flowing text layout.
- Some illustrations don't have any captions at all (e.g. Page 27/455).
- The keys are a bit strange in that they (a) have intermediate headings (e.g. Page 28/455) and (b) occasionally come with 3 leads in a single step (e.g. Step 1 in the key on Page 29/455). Handling keys with other than 2 leads per step surely is possible (also discussed with @mguidoti in an independent issue), but I'm not so sure about the intermediate headings ... the latter would require making the matching logic a good bit less selective, potentially incurring false positives.
- The keys are inconsistent in layout, just compare Page 29/455 to Page 30/455
- The numbering of figures and tables is prefixed with the chapter number and starts over at 1 at the start of every (genus oriented) chapter, which might necessitate some extensions to the figure and table citation handling logic.

**Comment by myrmoteras** [comment]
This this mean, it is doable? Shall we risk it?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by gsautter** [comment]
Doable is a lot ... all my points above pertain to fully automated batch processing of the whole book series. The worst that can happen (short of the PDFs not converting at all) is some additional manual correction work beyond what's usually required for QC.
So there is not a lot we'd really risk other than having to do a good few manual corrections. The treatments are surely worth getting out, they look nicely illustarted and rich in data, and as for the corrections, there is always the option to do some custom scripting if errors exhibit some schematics across multiple documents.

**Comment by gsautter** [comment]
I ran this PDF on the server now, which does have enough oomph to handle a PDF this size, and was able to reproduce and fix the decoding error (related to a character that does not render because normal fonts don't have a glyph for it) ... comes with next build.

**Comment by myrmoteras** [comment]
do you have the IMF somewher to contiue this work?

**Comment by gsautter** [comment]
The IMF sits on the server in my user home, yes. What's the best way of getting those 1.3 GB to you?

**Comment by gsautter** [comment]
The easiest might be to log in on the server with an SSH client and download it directly ...

**Comment by myrmoteras** [comment]
I agree- need tio figure out how to... that's what we actually discussed yesterday night. I see whether I can find my respective notes

**Comment by gsautter** [comment]
Well, not quite ... we talked about pushing a PDF into the server side batch yesterday.

In this case, I simply ran the conversion tool in my own user home on the machine, which is completely independent from out application back-end, server side batch, etc. I simply used the machine to run a command line tool.
